### ISSUE TITLE
Next.js production build doesn't remove certain types of comments (like: `//!`)

### ISSUE THREAD

**Issue by emzoumpo** [vercel/next.js#52600]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
I'm using Next.js 13.4.8 with the Pages Router, using Typescript.
```

### Link to the code that reproduces this issue or a replay of the bug

https://codesandbox.io/p/sandbox/restless-cache-f2vrkr

### To Reproduce

Add similar comments as depicted bellow on the description of the bug screenshots.

### Describe the Bug

Special comments like ones starting with: `//!` (that are still valid comments, like the ones from Better Comments) are not removed from the bundle when producing a production build.
![better-comments](https://github.com/vercel/next.js/assets/2103443/e560e512-34b2-4cc2-874b-1c0a24411ddb)

Comments starting with `//` are removed, as you can see in the pictures below (attaching the original React component and then the final one in the chunk) : 
<img width="1219" alt="Screenshot 2023-07-12 at 17 55 57" src="https://github.com/vercel/next.js/assets/2103443/77e25207-8ea8-4d6f-bd3a-ada8e9dee8a5">
<img width="1166" alt="Screenshot 2023-07-12 at 17 55 30" src="https://github.com/vercel/next.js/assets/2103443/168c1307-0704-4b1c-994f-bad58d3a0906">

### Expected Behavior

The generated chunks contain no comments

**Comment by emzoumpo** [comment]
@icyJoseph thanks a lot for pointing me out in the right direction, I opened an SWC issue instead: https://github.com/swc-project/swc/issues/7653

**Comment by balazsorban44** [comment]
Indeed, it's in SWC, so I'm closing this issue. When fixed there, we will update Next.js accordingly.

**Comment by emzoumpo** [comment]
@balazsorban44 fyi, this is now released with SWC [v1.3.71](https://github.com/swc-project/swc/milestone/428)
